common/src/node/ethereum/provider.rs

```diff
@@ -0,0 +1,22 @@
+use crate::EthError;
+use ethers::providers::{Http, Provider};
+use url::Url;
+
+static DEFIWALLETCORE_AGENTINFO: Option<&'static str> = option_env!("DEFIWALLETCORE_AGENTINFO");
+
+// urlinfo: url string of the node to connect to, "http://mynode:8545"
+// agentinfo: agent string for http header
+pub async fn get_ethers_provider(urlinfo: &str) -> Result<Provider<Http>, EthError> {
+    let url = Url::parse(urlinfo).unwrap();
+    let mut agentinfo = "defiwalletcore";
+    if let Some(v) = DEFIWALLETCORE_AGENTINFO {
+        agentinfo = v;
+    }
```

Comment:
nit: this could potentially be written as `let agentinfo = DEFIWALLETCORE_AGENTINFO.unwrap_or....`

Reply:
ok

Reply:
cosmos sdk also need to setup agent, i'll submit as separate issue.